packages/features/pbac/services/permission-check.service.ts

```diff
@@ -0,0 +1,59 @@
+import { MembershipRole } from "@calcom/prisma/enums";
+
+import type { PermissionString, Resource, Action } from "../types/permission-registry";
+import type { RoleService } from "./role.service";
+
+export class PermissionCheckService {
+  constructor(private roleService: RoleService) {}
+
+  async hasPermission(
+    membership: { role: MembershipRole; customRoleId: string | null },
```

Comment:
Require both their role and custom roleId if it exists for now 

Ideally this just takes in a roleId in the future but this is the cleanest way i saw to do a fall back for this POC